analyses/chromosomal-instability/02a-plot-chr-instability-heatmaps.Rmd

```diff
@@ -0,0 +1,300 @@
+---
+title: "Chromosomal Instability: Heatmaps"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+author: Candace Savonen for ALSF - CCDL
+date: 2020
+---
+
+This analysis evaluates chromosomal instability by using binned breakpoint counts 
+for SV and CNVdata that was formatted co-localized by individual samples in 
+`01-localization-of-breakpoints.Rmd`. 
+This notebook returns chromosomal break heatmaps in the `plots` directory.
+
+### Usage
+
+This notebook can be run via the command line from the top directory of the 
+repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/chromosomal-instability/02a-plot-chr-instability-heatmap.Rmd', 
+                              clean = TRUE)"
+```
+
+### Set Up
+
+```{r}
+# Set seed so heatmaps turn out the same
+set.seed(2020)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+### Directories and Files
+
+```{r}
+# Path to data directory
+data_dir <- file.path("..", "..", "data")
+scratch_dir <- file.path("..", "..", "scratch")
+
+# Path to output directory
+plots_dir <- "plots"
+
+# Create the hist_plots_dir  if it does not exist
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+Here's all the input files we will need: 
+
+```{r}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+binned_counts_files <- list.files("breakpoint-data",
+  pattern = "_binned_counts.tsv",
+  full.names = TRUE
+)
+```
+
+Output files:
+
+```{r}
+intersection_heatmap_file <- file.path(plots_dir, "intersection_breaks_heatmap.png")
+cnv_heatmap_file <- file.path(plots_dir, "cnv_breaks_heatmap.png")
+sv_heatmap_file <- file.path(plots_dir, "sv_breaks_heatmap.png")
+```
+
+Make a special function for making the heatmaps. 
+
+```{r}
+breaks_heatmap <- function(binned_counts_df,
+                           chrs,
+                           histologies,
+                           chr_colors,
+                           histologies_colors,
+                           col_fun) {
+  # A wrapper function for making a heatmap from the samples GenomicRanges list.
+  #
+  # Args:
+  # binned_counts_df: a data.frame with the binned counts for each sample.
+  # chrs: The chromosomes for each bin key.
+  # histologies: The histologies to biospecimen's key.
+  # chr_colors: A named vector to be used for coloring chromosomes.
+  # histologies_colors: A named vector to be used for coloring histologies.
+  # col_fun: a color key for the heatmap itself. Provided as a function.
+  #
+  # Returns:
+  # A heatmap of the chromosomal breaks
+
+  # Drop chr bin names
+  binned_counts_mat <- binned_counts_df %>%
+    dplyr::select(-chr_bin_names) %>%
```

Comment:
I figured out the other problem! You don't check here that the histologies and binned_counts have the same order. Since you sort the `histologies` data frame but not this, it turns out that they don't! 

So if you add in a step here to sort the binned count matrix, you should be good to go. 
The easiest way may be to simply use the order of the `histologies` rownames, since you are passing those in:

```suggestion
    dplyr::select(rownames(histologies)) %>%
```